### PR TITLE
Update k8s executor to follow specified routes when applying manifests

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/executor/kubernetes/canary_test.go [hunk 1]
```diff
@@ -175,11 +175,13 @@ func TestEnsureCanaryRollout(t *testing.T) {
 					}, nil)
 					return p
 				}(),
-				applier: func() provider.Applier {
-					p := kubernetestest.NewMockApplier(ctrl)
-					p.EXPECT().ApplyManifest(gomock.Any(), gomock.Any()).Return(fmt.Errorf("error"))
-					return p
-				}(),
+				applierGetter: &applierGroup{
```
Comment (knanao) [inline pkg/app/piped/executor/kubernetes/canary_test.go:178]: How about creating a mock for applierGetter?

Comment (nghialv) [reply pkg/app/piped/executor/kubernetes/canary_test.go:178]: We are checking the call of an internal object (default applier), so I think mock for applierGetter could not be helpful in this case.
Please correct me if I'm wrong.

Comment (knanao) [reply pkg/app/piped/executor/kubernetes/canary_test.go:178]: IMO, If you want to test whether the interface of applierGetter calls some methods or not, we should write tests for applierGroup to check it.
But I could also understand what you meant, it's up to you. 🙌

Comment (nghialv) [reply pkg/app/piped/executor/kubernetes/canary_test.go:178]: I got your point. You meant this test should only care about the applierGetter interface. If we want to test its internal calls that should be done in other tests, right?

Comment (nghialv) [reply pkg/app/piped/executor/kubernetes/canary_test.go:178]: @knanao OK. Let me do it together with this TODO https://github.com/pipe-cd/pipecd/pull/3817#issuecomment-1192405511 in another PR.

Comment (knanao) [reply pkg/app/piped/executor/kubernetes/canary_test.go:178]: Thank you 🙌

**Review by knanao — APPROVED** [review]

**Review by khanhtc1202 — APPROVED** [review]
Here you go

**Pull request merged**